src/autoquality/optimizer.py

```diff
@@ -148,16 +147,21 @@ def _configure_quality_control_from_params(pool: Pool) -> Pool:
         if 'ExamRequirement' in params and autoquality.exam_skill_id:
             exam_passing_skill_value = params['ExamRequirement']['exam_passing_skill_value']
             has_filter = False
+            if pool.filter and not isinstance(pool.filter, (FilterAnd, FilterOr)):
+                pool.filter = FilterAnd([pool.filter])
             if isinstance(pool.filter, FilterAnd):
-                for filter in pool.filter.and_:
+                for filter in pool.filter:
                     if isinstance(filter, Skill) and filter.key == autoquality.exam_skill_id:
                         filter.value = exam_passing_skill_value
                         has_filter = True
                         break
 
             if not has_filter:
                 exam_filter = (Skill(autoquality.exam_skill_id) >= exam_passing_skill_value)
-                pool.set_filter(pool.filter & exam_filter)
+                if pool.filter:
+                    pool.filter &= exam_filter
+                else:
+                    pool.filter = exam_filter
```

Comment:
It will be interpreted as `pool.filter &= (exam_filter if pool.filter else exam_filter)`. There will be a `TypeError` if `pool.filter is None`: 
```python 
TypeError: unsupported operand type(s) for &=: 'NoneType' and 'Skill'
```